### ISSUE TITLE
Why nuget restore generates Nuget folder under CI? 

### ISSUE THREAD

**Issue by devovercome** [NuGet/Home#6964]
Hi! 
I have a private git-server, which has post-receive hook.
This hook simply invokes kinda build.cmd
Which calls nuget restore inside solution folder:
```
MySln.sln
\src\Projects*.csproj
```

Somehow, under this CI routine, I am getting the **.nuget and NuGet folders**, so solution wont build.
Please, note that **it works as expected when** I call build.cmd directly (double-click), I am getting the packages folder, and solution will be built.

------
**Answer:**
I've got this problem **solved** when set **NUGET_PACKAGES** environment variable directly in my build script:
`SET NUGET_PACKAGES="%CD%\packages"`

See details in my last comment. 
Looks like hack, but I am fine with that.

PS: Please note: **no quote screening!** Otherwise wont work.

**Comment by zhili1208** [comment]
Could you share the error message you got during building solution? nuget will create .nuget folder as a global package cache, but it should not break solution build.

**Comment by devovercome** [comment]
@zhili1208 Thank you for reply! 

It doesnt break. 
Build is broken since CI-based build doesn't generate _packages_ folder.

I believe I've seen some log output (like GET nuget entries) and there was errors. But couldn't take a closer look, because of a lot work to do, so I just prefer make manual build anytime package added.
 
Hang on for further investigation.

**Comment by devovercome** [comment]
@zhili1208 Thank you for the point!

Since I've learned that .nuget folder is a global package cache, but has been created under local folder of my project, I figured that there is something wrong under network account.

I've took a time a read logs carefully, they said something like
_access to path C:\Windows\system32\config\systemprofile_

Hello! This is wrong path!

**My solution**
Is based on @emgarten answer [here](https://github.com/NuGet/Home/issues/4277#issuecomment-272538893) - I just set a **NUGET_PACKAGES** variable to exact package folder where I expect to have packages. There should be a better way but I **so** used to override things, alas.

I will fill initial post with my solution and close issue since it is a question and its answered!

Again, thanks in advance. I eager to learn nuget better to expect the unexpected!
